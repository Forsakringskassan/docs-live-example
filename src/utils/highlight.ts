import { type Options } from "prettier";
import prettier from "prettier/standalone";
import htmlPlugin from "prettier/parser-html";
import hljs from "highlight.js/lib/core";
import html from "highlight.js/lib/languages/xml";

hljs.registerLanguage("html", html);

const prettierConfig: Options = {
    parser: "html",
    plugins: [htmlPlugin],
    singleQuote: false,
    arrowParens: "always",
    tabWidth: 4,
    printWidth: 80,
};

/**
 * @internal
 */
export async function highlight(code: string): Promise<string> {
    /* eslint-disable-next-line import/no-named-as-default-member -- technical
     * debt, without actually checking this is probably needed for compatibility
     * with prettier 2 */
    const formatted = await prettier.format(code, prettierConfig);
    const { value } = hljs.highlight(formatted, { language: "html" });
    return `<code class="hljs lang-html" tabindex="0">${value}</code>`;
}
