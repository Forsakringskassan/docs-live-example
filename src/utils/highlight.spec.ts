import { highlight } from "./highlight";

it("should be formatted and highlighted", async () => {
    expect.assertions(1);
    const html = `<div class="my-class"><div><p>Lorem ipsum dolor sit amet</p></div></div>`;
    expect(await highlight(html)).toMatchSnapshot();
});
