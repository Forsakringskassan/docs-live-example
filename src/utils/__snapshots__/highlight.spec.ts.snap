// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be formatted and highlighted 1`] = `
"<code class="hljs lang-html" tabindex="0"><span class="hljs-tag">&lt;<span class="hljs-name">div</span> <span class="hljs-attr">class</span>=<span class="hljs-string">&quot;my-class&quot;</span>&gt;</span>
    <span class="hljs-tag">&lt;<span class="hljs-name">div</span>&gt;</span><span class="hljs-tag">&lt;<span class="hljs-name">p</span>&gt;</span>Lorem ipsum dolor sit amet<span class="hljs-tag">&lt;/<span class="hljs-name">p</span>&gt;</span><span class="hljs-tag">&lt;/<span class="hljs-name">div</span>&gt;</span>
<span class="hljs-tag">&lt;/<span class="hljs-name">div</span>&gt;</span>
</code>"
`;
